### ISSUE TITLE
[Trevi] Entity body와 Middleware BodyParser

### ISSUE THREAD

**Issue by Yoseob** [D2CAMPUS-FEST/2015#14]
요청으로 들어온 Entity Body에 이미지나 파일을 메모리 문제 때문에 Request 객체의 Body에 바로 넣지 않고 파일처리를 통해서 파일의 경로를 참조하는 형식으로 구현할 생각입니다. 이에 관해서 필드의 Content-type으로 구분지어 Request의 객체에 들어갈지 파일에 들어갈지 구분하고, Middleware bodyparser 에서  파일을 읽어 파싱을 할 생각입니다. 관련 자료도 부족하고 
어떻게 하면 좋은 방식인지, 다른 framework에서는 요청된 이미지를 어떤식으로 저장하고 파싱 하는지 궁금합니다. 

https://github.com/Yoseob/Trevi/blob/master/Trevi/Middleware/BodyParser.swift#L100-L120

**Comment by MyeongKim** [comment]
Node.js 에서 파일 업로드를 Nginx 로 처리하는 방식입니다. 
Node.js 서버 코드를 거치지 않고 Nginx의 업로드 설정을 통해 지정된 디렉토리에 저장됩니다.

https://coderwall.com/p/swgfvw/nginx-direct-file-upload-without-passing-them-through-backend

**Comment by Yoseob** [comment]
정보 감사합니다! 그런데 
Node.js가  Nginx 기반이 아니지 않나요?

**Comment by MyeongKim** [comment]
@Yoseob 
네. 제가 Node.js 를 언급한 이유는 다른 framework 에서 이미지 업로드를 어떻게 처리하는지 설명하기 위해 말씀드린 것입니다. 

정확하게는 Nginx의 기능이지만 제가 다른 framework의 경우에 대해서 잘 모르기 때문에 Node.js와 Nginx를 같이 말씀드렸습니다.

**Comment by Yoseob** [comment]
@mingkim 네. 코멘트 감사합니다!!
